solutions/USACO feb17-circlecross.mdx

```diff
@@ -2,223 +2,47 @@
-		return b ? "true" : "false"; 
-	#else 
-		return ts((int)b);
-	#endif
-}
-tcT> str ts(complex<T> c) { 
-	stringstream ss; ss << c; return ss.str(); }
-str ts(V<bool> v) {
-	str res = "{"; F0R(i,sz(v)) res += char('0'+v[i]);
-	res += "}"; return res; }
-template<size_t SZ> str ts(bitset<SZ> b) {
-	str res = ""; F0R(i,SZ) res += char('0'+b[i]);
-	return res; }
-tcTU> str ts(pair<T,U> p);
-tcT> str ts(T v) { // containers with begin(), end()
-	#ifdef LOCAL
-		bool fst = 1; str res = "{";
-		for (const auto& x: v) {
-			if (!fst) res += ", ";
-			fst = 0; res += ts(x);
-		}
-		res += "}"; return res;
-	#else
-		bool fst = 1; str res = "";
-		for (const auto& x: v) {
-			if (!fst) res += " ";
-			fst = 0; res += ts(x);
-		}
-		return res;
+**Time Complexity:** $\mathcal{O}(N^4)$ (with $N$ being the length of the input string which is $52$)
 
-	#endif
-}
-tcTU> str ts(pair<T,U> p) {
-	#ifdef LOCAL
-		return "("+ts(p.f)+", "+ts(p.s)+")"; 
-	#else
-		return ts(p.f)+" "+ts(p.s);
-	#endif
-}
+<LanguageSection>
 
-// OUTPUT
-tcT> void pr(T x) { cout << ts(x); }
-tcTUU> void pr(const T& t, const U&... u) { 
-	pr(t); pr(u...); }
-void ps() { pr("\n"); } // print w/ spaces
-tcTUU> void ps(const T& t, const U&... u) { 
-	pr(t); if (sizeof...(u)) pr(" "); ps(u...); }
+<CPPSection>
 
-// DEBUG
-void DBG() { cerr << "]" << endl; }
-tcTUU> void DBG(const T& t, const U&... u) {
-	cerr << ts(t); if (sizeof...(u)) cerr << ", ";
-	DBG(u...); }
-#ifdef LOCAL // compile with -DLOCAL, chk -> fake assert
-	#define dbg(...) cerr << "Line(" << __LINE__ << ") -> [" << #__VA_ARGS__ << "]: [", DBG(__VA_ARGS__)
-	#define chk(...) if (!(__VA_ARGS__)) cerr << "Line(" << __LINE__ << ") -> function(" \
-		 << __FUNCTION__  << ") -> CHK FAILED: (" << #__VA_ARGS__ << ")" << "\n", exit(0);
-#else
-	#define dbg(...) 0
-	#define chk(...) 0
-#endif
+## C++ Implementation
 
-void setPrec() { cout << fixed << setprecision(15); }
-void unsyncIO() { cin.tie(0)->sync_with_stdio(0); }
-// FILE I/O
-void setIn(str s) { freopen(s.c_str(),"r",stdin); }
-void setOut(str s) { freopen(s.c_str(),"w",stdout); }
-void setIO(str s = "") {
-	unsyncIO(); setPrec();
-	// cin.exceptions(cin.failbit); 
-	// throws exception when do smth illegal
-	// ex. try to read letter into int
-	if (sz(s)) setIn(s+".in"), setOut(s+".out"); // for USACO
-}
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
 
 int main() {
-	setIO("circlecross");
-	str s; re(s);
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	freopen("circlecross.in","r",stdin);
+	freopen("circlecross.out","w",stdout);
+	string s; cin >> s;
 	int ans = 0;
-	F0R(a,sz(s)) FOR(b,a+1,sz(s)) FOR(c,b+1,sz(s)) FOR(d,c+1,sz(s))
-		ans += s[a] == s[c] && s[b] == s[d];
-	ps(ans);
+	for (int a = 0; a < s.size(); ++a) 
+		for (int b = a+1; b < s.size(); ++b) 
+			for (int c = b+1; c < s.size(); ++c) 
+				for (int d = c+1; d < s.size(); ++d) 
+					ans += s[a] == s[c] && s[b] == s[d];
+	cout << ans;
+	return 0;
 }
-```
\ No newline at end of file
+```
```

Comment:
Add alternate O(N^2) solution?

Reply:
maybe

Reply:
I just added the O(N^2) solution for the 2nd solution. Hope it might help

Reply:
Thanks!